.github/workflows/linux.yml

```diff
@@ -24,15 +24,15 @@ env:
 
 jobs:
 
-  ########################
-  # linux-release-serial #
-  ########################
+  ##################
+  # release-serial #
+  ##################
 
-  linux-release-serial:
+  release-serial:
```

Comment:
I dont understand why you use a matrix strategy for runners for this job and for `debug-parallel` and `debug-parallel-tpetra` (every job that contains `container: image:`). These jobs are executed inside the container we specify in `image: ...`. So which runner we use outside the container doesnt matter. Dont you duplicate these runs with the matrix strategy you propose here?

I would understand if you want to use a matrix strategy for different images, but as you mention we dont have different images (or at least dont use them at the moment).

Reply:
> So which runner we use outside the container doesnt matter. Dont you duplicate these runs with the matrix strategy you propose here?

I admit that my understanding of Docker containers is limited. You are right -- it makes no sense to run a container image (see also [here](https://github.com/orgs/community/discussions/25534#discussioncomment-3248244)) on two different host OSs. Until we have a `noble` docker container, my suggestion does not make sense.